src/games/connect_four.rs

```diff
@@ -96,6 +99,17 @@ macro_rules! check_offsets {
 	};
 }
 
+fn bitmask_to_count(mask: Mask<i8, 4>) -> u8 {
+	match mask.to_bitmask() {
+		0b0000 => 0,
+		0b0001 | 0b0010 | 0b0100 | 0b1000 => 1,
+		0b0011 | 0b0101 | 0b0110 | 0b1001 | 0b1010 | 0b1100 => 2,
+		0b0111 | 0b1011 | 0b1101 | 0b1110 => 3,
+		0b1111 => 4,
+		_ => unsafe { unreachable_unchecked() },
+	}
+}
```

Comment:
just noticed something i didn't earlier. this can be replaced with `u64::count_ones` (which uses intrinsics::ctpop internally, and is highly performant).
```suggestion
#[inline(always)]
fn bitmask_to_count(mask: Mask<i8, 4>) -> u8 {
	mask.to_bitmask().count_ones()
```

Reply:
That's what I had originally, but it uses too many instructions.

```asm
; fn bitmask_to_count(mask: Mask<i8, 4>) -> u8 {
     movd    xmm0, dword, ptr, [rcx]
     pmovmskb ecx, xmm0
     mov     eax, ecx
     shr     al
     and     al, 85
     sub     cl, al
     mov     eax, ecx
     and     al, 51
     shr     cl, 2
     and     cl, 51
     add     cl, al
     mov     eax, ecx
     shr     al, 4
     add     al, cl
     and     al, 15
; }
 ret
```

The current code is much smaller and avoids using the same registry variable in many consecutive instructions:
```asm
; fn bitmask_to_count(mask: Mask<i8, 4>) -> u8 {
     movd    xmm0, dword, ptr, [rcx]
     pmovmskb eax, xmm0
; match mask.to_bitmask() {
 movzx   eax, al
 lea     rcx, [rip, +, switch.table._ZN8skyra_ai5games12connect_four16bitmask_to_count17h49140c22359dd8aeE]
 movzx   eax, byte, ptr, [rax, +, rcx]
; }
 ret
```

Reply:
that's strange. well, feel free to resolve. we should probably add comments to these explaining why they are how they are, so future onlookers don't have the same questions

Reply:
Done in d611caf